### ISSUE TITLE
[Issue] `azd env list --output json` returns non-valid JSON output when later version of `azd` is available

### ISSUE THREAD

**Issue by weikanglim** [Azure/azure-dev#683]
- [x] Make sure you've installed the latest version using [instructions in the wiki](../wiki/install)

**Output from `azd version`**
Run `azd version` and copy and paste the output here:
azd version 0.1.0-beta.5 (commit 7bd25dd6f1b1899abf63b5f6f36d268a465183e4)

**Output from `az version`** 
Run `az version` and copy and paste the output here (minimum required version is 2.38.0): 
{
  "azure-cli": "2.38.0",
  "azure-cli-core": "2.38.0",    
  "azure-cli-telemetry": "1.0.6",
  "extensions": {
    "azure-devops": "0.22.0",    
    "resource-graph": "2.1.0"    
  }
}

**Describe the bug**
When a later version of `azd` is available, non-json output is being sent to standard output. This actually breaks VSCode local development capabilities, since the list of environments cannot be parsed, which means the `.env` file cannot be loaded.

To work around, one simply needs to update the `azd` version.

**To Reproduce**
In `pwsh`:
1. Install an older version of `azd`: `Invoke-RestMethod 'https://aka.ms/install-azd.ps1' -OutFile 'install-azd.ps1' && ./install-azd.ps1 -Version '0.1.0-beta.5`
2. Run `json = (azd env list --output json) | ConvertFrom-Json`
3. Examine the error message:
> ConvertFrom-Json: Conversion from JSON failed with error: Additional text encountered after finished reading JSON content: w. Path '', line 8, position 0.      

Output of `azd env list --output json`:
```
PS D:\repos\azd-sources\demo-todo-python> azd env list --output json
[
  {
    "Name": "weilim-demo-todo-python-123",
    "IsDefault": false,
    "DotEnvPath": "D:\\repos\\azd-sources\\demo-todo-python\\.azure\\weilim-demo-todo-python-123\\.env"
  }
]
warning: your version of azd is out of date, you have 0.1.0-beta.5 and the latest version is 0.2.0-beta.1

To update to the latest version, run:
powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' 
| Invoke-Expression"
```

**Expected behavior**
`azd env list --output json` should always print out valid JSON. 

**Environment**
Information on your environment:
    * Language name and version
    * IDE and version : [e.g. Visual Studio 16.3]

**Additional context**
Related to #86

**Comment by karolz-ms** [comment]
People who upgrade to latest version will not suffer from problems with VS Code I presume?

Is this warning emitted to `stdout` or `stderr`?

**Comment by weikanglim** [comment]
That's correct. The fix is to follow the message and upgrade `azd` to latest.

The warning is being emitted to `stdout`:
```pwsh
PS D:\repos\azd-sources\demo-todo-python> azd env list --output json 1>azd.out && Get-Content azd.out
[
  {
    "Name": "weilim-demo-todo-python-123",
    "IsDefault": false,
    "DotEnvPath": "D:\\repos\\azd-sources\\demo-todo-python\\.azure\\weilim-demo-todo-python-123\\.env"
  }
]
warning: your version of azd is out of date, you have 0.1.0-beta.5 and the latest version is 0.2.0-beta.1

To update to the latest version, run:
powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' 
| Invoke-Expression"
```

**Comment by ellismg** [comment]
Fixed with #685
